src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java

```diff
@@ -111,9 +119,18 @@ public void execute() throws MojoExecutionException {
             getLog().info("About to generate Cucumber report.");
             Reportable report = reportBuilder.generateReports();
 
-            if (checkBuildResult && (report == null || report.getFailedSteps() > 0)) {
-                throw new MojoExecutionException("BUILD FAILED - Check Report For Details");
-            }
+			if (checkBuildResult && null != report && report.getScenarios() > report.getPassedScenarios()) {
+				float totalScenarios = report.getScenarios();
+				float passedScenarios = report.getPassedScenarios();
+				getLog().info("Total Scenarios : " + totalScenarios + " , Passed Scenarios :  " + passedScenarios);
+				float percent = ((passedScenarios / totalScenarios) * 100);
+				getLog().info(
+						"Test Pass percent : " + percent + "% , PassPercentThreshold : " + passPercentThreshold + "%");
+				if (passPercentThreshold > percent) {
+					throw new MojoExecutionException("BUILD FAILED - Check Report For Details");
```

Comment:
can you pass the information about details instead of asking to check the report manually?